File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void printrow(int row, int height)
{
    //setup the leading spaces
    int spaces = 1;
    while (spaces <= (height - row))
    {
        printf(" ");
        spaces++;
    }

    //setup the left side hash
    int hashes = 1;
    while (hashes <= row)
    {
        printf("#");
        hashes++;
    }

    //setup the middle spaces
    printf("  ");

    //setup the right side hash
    hashes = 1;
    while (hashes <= row)
    {
        printf("#");
        hashes++;
    }

    //end the line and move to the next line
    printf("\n");
}


int main(void)
{
    // Get the user input for pyramid height
    int height = 0;
    while (height < 1 || height > 8)
    {
        height = get_int("Height: ");
    }

    //print one row at a time
    //until the height is reached
    int i = 1;
    while (i <= height)
    {
        printrow(i, height);
        i++;
    }
}

File: sentimental-mario-more/mario.py
from cs50 import get_int


# function to print the leading spaces plus the trailing hash
def printrow(height, n):
    # handle the base case
    if height == 0:
        return False

    # recursively print the half pyramid
    print(" " * (height - 1) + "#" * n + "  " + "#" * n)
    printrow(height - 1, n + 1)

    return True


# get the user input and make sure it is valid
while(True):
    height = get_int("Height: ")

    # check for height validity
    if int(height) < 1 or int(height) > 8:
        print("Height must be between 1 and 8")
    elif int(height) > 0 and int(height) < 9:
        printrow(height, 1)
        break